### ISSUE TITLE
Proposal to add a few types and properties to express energy efficiency of products

### ISSUE THREAD

**Issue by alex-jansen** [schemaorg/schemaorg#2670]
**Context** - This is a proposal from Google based on our experience consuming schema.org Product markup and working with similar data from online merchants. If it were accepted, it would make it easier for us and others to understand energy efficiency properties of products.

### **Introduction**

Many products advertised or sold on the web consume energy and countries around the world have regulations related to labeling such products with energy efficiency information, both online and in stores. An example of such a regulation is [EU directive 2017/1369](https://eur-lex.europa.eu/eli/reg/2017/1369/oj), which covers energy labeling for [various categories of products](https://ec.europa.eu/energy/topics/energy-efficiency/energy-efficient-products/list-regulations-product-groups-energy-efficient-products_en?redir=1). Another example is the [Energy labeling rule](https://www.ftc.gov/enforcement/rules/rulemaking-regulatory-reform-proceedings/energy-water-use-labeling-consumer) under the Energy Policy and Conservation Act (EPCA) in the US. In addition to these regulations, consumers in general are increasingly interested in the energy efficiency of products they are researching online.

The energy efficiency characteristics of a product are typically expressed through a combination of one or more of the following elements:

*   An energy efficiency rating, which could be:
    *   A flag indicating that the product meets a particular energy efficiency standard (for example, the [Energy Star](https://en.wikipedia.org/wiki/Energy_Star) rating in the US), or: 
    *   A value out of a range of energy efficiency classes (https://en.wikipedia.org/wiki/European_Union_energy_label) in the EU. Each such class indicates a different level of energy efficiency (high to low). Different products have a different range (spectrum) of classes, for example [A+++, …, D] for dishwashers or [A,...,G] for [A++,...,E] for lamps.
*   An energy efficiency label (image) in a standardized format. Examples: [EU](https://en.wikipedia.org/wiki/European_Union_energy_label#/media/File:Energy_label_2010.svg), [NZ](https://www.eeca.govt.nz/assets/Images-EECA/logos-labels/energy-rating-label-large.png), [US](https://www.consumer.ftc.gov/sites/default/files/pictures/0072-energy-guide-label.jpg).
*   A datasheet (or "fiche"), specifying details such as energy consumption, water consumption, loudness, and many others. The required line items on such a datasheet are typically different for different types of products (see examples of EU fiche requirements for [air conditioners](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX:32011R0626#L_2011178EN.01006401) and [wine coolers](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX:32010R1060#L_2010314EN.01003501)) and must be supplied in a specific order.

Currently, Schema.org provides no explicit support for expressing energy efficiency properties of products. Although the /_PropertyValue_ class could be used, we believe allowing an explicit and structured way of expressing energy consumption characteristics by adding a small amount of vocabulary would benefit the ecosystem.

### **Proposal**

Our proposal is to add a new class, for example /_EnergyConsumptionDetails_, that would define the energy consumption characteristics of a product. This class would then be used by a new property _/energyConsumptionDetails_ under /_Product._

Properties under this new class could include, for example:

*   _energyEfficiencyRating_, a value from a new  _/EnergyEfficiencyRating_ enumeration type, containing values such as "EnergyStar", "A+++", "G", etc.
*   _energyEfficiencyRatingRange_, a value from a new class /_EnergyEfficiencyRatingRangeSpecification_ to specify a range (from-to) of energy efficiency ratings.
*   _energyLabelImage_, a URL to an image of an energy efficiency label.
*   _datasheet_, for which we could consider using the existing _/ItemList_ class to represent a list of line items on the datasheet, each item being a key-value pair which could be modeled using _/PropertyValue_.

### **Some Open Questions**

*   We could consider adding an explicit regional or country identifier to identify the specific locale (EU, US, …) a particular _/EnergyConsumptionDetails_ instance belongs to.
*   Adding more structure to the datasheet (as an alternative to using key-value pairs) can be considered but for the EU alone this would lead to a very large (over 200) number of specific properties for over 25 product groups that would need to be modeled explicitly.

**Comment by danbri** [comment]
Thanks for the great writeup :) 

Somewhat relevant here - @lucy-kind recently led some cleanup of our structure around Enumerations and enumerated values, to make tidier use of a type hierarchy. We've applied that idea to national code lists too, e.g. take a look at 
 https://schema.org/NonprofitType and https://schema.org/NLNonprofitType in https://github.com/schemaorg/schemaorg/blob/main/data/ext/pending/issue-2543.ttl - in theory we can now collect such definitions from a variety of countries / territories / regions. There will doubtless be open questions on how exactly to handle this in markup but I think we have a stronger foundation for regionalized enumerations now (also thanks @rjw for the implementation).

**Comment by alex-jansen** [comment]
It is an excellent idea to use country (or region) specific enum types for energy efficiency ratings, since for example the EU has a very different energy efficiency scale (A+++ - G, depending on the product category) than Korea or China (1-5). 

We could introduce a new enumeration /EnergyEfficiencyClass with sub-enum types /EUEnergyEfficiencyClass, /KEAEnergyEfficiencyClass (for Korea, also used in other countries such as Cambodia), etc.

A complete example for a /energyConsumptionDetails property under a /Product could then look as follows:

```

  "energyConsumptionDetails": {
    "@type": "EnergyConsumptionDetails",
    "energyClass": "https://schema.org/EUEnergyEfficiencyClassA1Plus",
    "energyScaleMin": "https://schema.org/EUEnergyEfficiencyClassD",
    "energyScaleMax": "https://schema.org/EUEnergyEfficiencyClassA1Plus",
    "energyLabelImage": "https://www.example.com/energy_label_12345.jpg",
    "datasheet": {
      "@type": "ItemList",
      "itemListOrder": "http://schema.org/ItemListOrderAscending",
      "itemListElement": [
        {
          "@type": "ListItem",
          "position": 1,
          "item": {
            "@type": "PropertyValue",
            "name": "supplier",
            "value": "my tv brand"
          }
        },
        {
          "@type": "ListItem",
          "position": 2,
          "item": {
            "@type": "PropertyValue",
            "name": "model",
            "value": "AA-12345Z"
          }
        },
        {
          "@type": "ListItem",
          "position": 3,
          "item": {
            "@type": "PropertyValue",
            "name": "energy efficiency class",
            "value": "A+"
          }
        },
        {
          "@type": "ListItem",
          "position": 4,
          "item": {
            "@type": "PropertyValue",
            "name": "visible screen diagonal",
            "value": "189 cm (75 inch)"
          }
        },
        {
          "@type": "ListItem",
          "position": 5,
          "item": {
            "@type": "PropertyValue",
            "name": "power consumption (in operation)",
            "value": "361 W"
          }
        },
        {
          "@type": "ListItem",
          "position": 6,
          "item": {
            "@type": "PropertyValue",
            "name": "energy consumption",
            "value": "543 kWh per year, based on the power consumption of the television operating 4 hours per day for 365 days. The actual energy consumption will depend on how the television is used."
          }
        },
        {
          "@type": "ListItem",
          "position": 7,
          "item": {
            "@type": "PropertyValue",
            "name": "standby/off power consumption",
            "value": "0,50 W"
          }
        },
        {
          "@type": "ListItem",
          "position": 8,
          "item": {
            "@type": "PropertyValue",
            "name": "screen resolution",
            "value": "7680 x 4210 pixels (H x V)"
          }
        }
      ]
    }
  
```

The above example defines the energy efficiency class in the EU ("A+" in this case, using a proposed enum value /EUenergyEfficiencyClassA1Plus) as well as the scale for the particular product category (A+ - D for televisions). The example also gives a link to an image of an energy efficiency label ([example](http://ec.europa.eu/avservices/avs/files/video6/repository/prod/photo/store/store2/5/P039725-101020.jpg)) as well as the product fiche (datasheet) following the EU regulation ([link](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX%3A32010R1062#d1e32-70-1)).

Another example, this time modeling the EnergyStar rating used in the US and many other countries could look as follows:

```
  "energyConsumptionDetails": {
    "@type": "EnergyConsumptionDetails",
    "energyClass": "https://schema.org/EnergyStarCompliant,
    "energyLabelImage": "https://www.example.com/energystar_label.jpg"
  }
```



A topic for further discussion could be to support multiple energy label images and product data sheets per /energyConsumptionDetails instance, with each image and data sheet tagged with the language and/or country it applies to:
- Different countries may have different labeling requirements, even if using the same energy rating standard.
  * ([example](https://www.nrcan.gc.ca/sites/www.nrcan.gc.ca/files/energy/energystar/EX_LC_COL_WD_E_2020.jpg))
  * ([literature review](https://ethz.ch/content/dam/ethz/special-interest/gess/economics-dam/documents/Projekts/EnergyLabelReview.pdf))
- Multiple Product fiches (datasheets) in different languages are common in the EU

**Comment by danbri** [comment]
I've committed a design that @alex-jansen (from Google Shopping) and I have been working on. 

It is fairly straightforward: a type representing EnergyConsumptionDetails, a property linking it from a Product, a property of the new type with enumerated values. Also an optional (EU-oriented) pair of properties of EnergyConsumptionDetails indicating maximum and minimum enumerated values. The enumerated structures make use of subtyping, so that there is a different Enumeration structure for Energy Star, and for EU's system. We have gone with Energy Star as an enumeration rather than a simple boolean, for consistency, since there are other systems (e.g. in China, Korea) that are also enumerated, and it seemed better to have a common pattern. The max/min construction is for use in the EU scheme, where it is often required to indicate the applicable range for a given product. For it to make sense, the enumeration needs to be understood as ordered; this is not currently explicit in schema.org's enumeration system can be concluded from reading the specific definitions that we give.

Background reading:

* https://en.wikipedia.org/wiki/China_Energy_Label
* https://www.energy.or.kr/renew_eng/energy/appliances/labeling.aspx
* https://en.wikipedia.org/wiki/Energy_Star
* http://www.kemco.or.kr/nd_file/kemco_eng/KoreaEnergyStandards&Labeling.pdf
* https://ethz.ch/content/dam/ethz/special-interest/gess/economics-dam/documents/Projekts/EnergyLabelReview.pdf

**Comment by danbri** [comment]
Draft staged: https://webschemas.org/EnergyConsumptionDetails

We should probably adopt @lucy-kind 's proposal and rename the enumeration to end in *Enumeration, but that leaves the question of whether to retain use of 'category' in hasEnergyEfficiencyCategory (the actual domain terminology seems to use 'class' heavily but that's a poor fit for us as 'class' at schema.org is the same thing as a type, i.e. a fundamental modeling primitive).

**Comment by alex-jansen** [comment]
This was published in release #10.0 (https://www.schema.org/EnergyConsumptionDetails)

**Comment by DamonHD** [comment]
For this to make sense for the EU labels at least don't we also need to know the year that the label was issued?

I had two new appliances that were simultaneously issued with A++ and D labels (or similar) because of the rebasing for those categories.

It might also make sense to have slots for at least the annual energy consumption and an optional URL for the full fiche/document?

Or do I need to read this thread from the top properly to find out why I'm wrong?  B^>

Rgds

Damon
